### PR TITLE
mpvScripts: use stdenvNoCC and misc. cleanup

### PR DESCRIPTION
**Pull request by lunik1**
###### Motivation for this change

A few cleanup items for mpv plugins:

1. Use `stdenvNoCC` when building, as a C compiler is not needed (`mpvScripts.mpdris` notwithstanding)
2. Drop the unused `fetchurl` dependency in `mpvScripts.autoload`
3. Set the license of `mpvScripts.convert` to `unfree` since the upstream gist has no license

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
While we're at it, we probably should mark convert as broken as well: #113202.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119968` at eee5a743 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>mpvScripts.mpvacious</li>
    <li>mpvScripts.sponsorblock</li>
    <li>mpvScripts.thumbnail</li>
  </ul>
</details>

**Comment by lunik1** [comment]
4. Mark `mpvScripts.convert` as broken (#113202)

File: pkgs/applications/video/mpv/scripts/convert.nix [hunk 1]
```diff
@@ -30,14 +30,16 @@ stdenv.mkDerivation {
...
     description = "Convert parts of a video while you are watching it in mpv";
     homepage = "https://gist.github.com/Zehkul/25ea7ae77b30af959be0";
-    maintainers = [ lib.maintainers.Profpatsch ];
+    maintainers = [ maintainers.Profpatsch ];
     longDescription = ''
       When this script is loaded into mpv, you can hit Alt+W to mark the beginning
       and Alt+W again to mark the end of the clip. Then a settings window opens.
     '';
+    license = licenses.unfree;
+    broken = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mpv/scripts/convert.nix:42]:
```suggestion
    # script crashes mpv. See https://github.com/NixOS/nixpkgs/issues/113202
    broken = true;
```

**Review by SuperSandro2000 — APPROVED** [review]
other than that LGTM

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**